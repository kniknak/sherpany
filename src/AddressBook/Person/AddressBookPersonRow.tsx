import * as React from "react";
import TableCell from "@material-ui/core/TableCell";
import { FetchAddressBookUser } from "../redux/types";
import MoreIcon from "@material-ui/icons/MoreVert";
import IconButton from "@material-ui/core/IconButton";
import Avatar from "@material-ui/core/Avatar";

export interface AddressBookPersonRowProps {
    person: FetchAddressBookUser
    onOpen: () => void
}

export class AddressBookPersonRow extends React.PureComponent<AddressBookPersonRowProps> {
    render() {
        const { person, onOpen } = this.props;

        return (
            <>
                <TableCell>
                    <Avatar
                        alt={person.name.first + " " + person.name.last}
                        src={person.picture.thumbnail}
                    />
                </TableCell>
                <TableCell>{person.name.first}</TableCell>
                <TableCell>{person.name.last}</TableCell>
                <TableCell>{person.login.username}</TableCell>
                <TableCell>{person.email}</TableCell>
                <TableCell>
                    <IconButton
                        color="inherit"
                        onClick={onOpen}
                    >
                        <MoreIcon/>
                    </IconButton>
                </TableCell>
            </>
        );
    }
}