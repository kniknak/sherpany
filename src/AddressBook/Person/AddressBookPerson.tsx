import * as React from "react";
import { FetchAddressBookUser } from "../redux/types";
import { AddressBookPersonRow } from "./AddressBookPersonRow";
import TableRow from "@material-ui/core/TableRow";

import * as styles from "./AddressBookPerson.module.scss"

export interface AddressBookPersonProps {
    person: FetchAddressBookUser
    isShown: boolean
    onOpen: (person: FetchAddressBookUser) => void
}

export class AddressBookPerson extends React.PureComponent<AddressBookPersonProps> {
    private onOpen = () => this.props.onOpen(this.props.person)

    render() {
        const { person, isShown } = this.props;

        return (
            <TableRow className={isShown ? undefined : styles.hidden}>
                <AddressBookPersonRow
                    person={person}
                    onOpen={this.onOpen}
                />
            </TableRow>
        );
    }
}