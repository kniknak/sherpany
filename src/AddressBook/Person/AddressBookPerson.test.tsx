import React from "react";
import { shallow } from "enzyme";
import { AddressBookPerson, AddressBookPersonProps } from "./AddressBookPerson";
import TableRow from "@material-ui/core/TableRow";

import * as styles from "./AddressBookPerson.module.scss"
import { AddressBookPersonRow } from "./AddressBookPersonRow";

describe("AddressBook", () => {
    describe("Person", () => {
        describe("AddressBookPerson", () => {
            const defaultProps: AddressBookPersonProps = {
                person: {
                    name: {
                        first: "name.first",
                        last: "name.last",
                    },
                    location: {
                        street: {
                            name: "location.street.name",
                            number: 1,
                        },
                        city: "location.city",
                        state: "location.state",
                        postcode: "location.postcode",
                    },
                    email: "email",
                    login: {
                        username: "login.username",
                    },
                    phone: "phone",
                    cell: "cell",
                    id: {
                        value: "id.value",
                    },
                    picture: {
                        thumbnail: "picture.thumbnail",
                    },
                },
                isShown: true,
                onOpen: () => undefined,
            };

            it("renders", () => {
                const subject = shallow(<AddressBookPerson {...defaultProps}/>);

                expect(subject).toMatchSnapshot()
            });

            it("renders hidden", () => {
                const props = {
                    ...defaultProps,
                    isShown: false,
                }
                const subject = shallow(<AddressBookPerson {...props}/>);

                expect(subject.find(TableRow).prop("className")).toBe(styles.hidden)
            });

            it("opens modal", () => {
                const props = {
                    ...defaultProps,
                    onOpen: jest.fn()
                }
                const subject = shallow(<AddressBookPerson {...props}/>);

                subject.find(AddressBookPersonRow).prop("onOpen")()

                expect(props.onOpen).toHaveBeenLastCalledWith(props.person)
            });
        });
    });
});