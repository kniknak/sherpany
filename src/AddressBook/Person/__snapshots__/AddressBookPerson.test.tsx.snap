// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddressBook Person AddressBookPerson renders 1`] = `
<WithStyles(ForwardRef(TableRow))>
  <AddressBookPersonRow
    onOpen={[Function]}
    person={
      Object {
        "cell": "cell",
        "email": "email",
        "id": Object {
          "value": "id.value",
        },
        "location": Object {
          "city": "location.city",
          "postcode": "location.postcode",
          "state": "location.state",
          "street": Object {
            "name": "location.street.name",
            "number": 1,
          },
        },
        "login": Object {
          "username": "login.username",
        },
        "name": Object {
          "first": "name.first",
          "last": "name.last",
        },
        "phone": "phone",
        "picture": Object {
          "thumbnail": "picture.thumbnail",
        },
      }
    }
  />
</WithStyles(ForwardRef(TableRow))>
`;
